_playwright-tests/UI/CustomRepoPagination.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { bulkCreateRepos } from './helpers/createRepositories';
+import { closePopupsIfExist } from './helpers/helpers';
+
+const repoNamePrefix = 'custom_repo-pagination';
+
+test.describe('Custom repositories pagination', () => {
+  test('Test pagination on custom repo page', async ({ page }) => {
+    await test.step('Delete any pagination test repos that exist', async () => {
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+      await page.waitForTimeout(1500); // Wait for soft delete
+    });
+    await test.step('Populate the custom repo table', async () => {
+      await bulkCreateRepos(page, 12); // 12 counting from 1
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await page.getByPlaceholder(/^Filter by name.*$/).fill(repoNamePrefix);
+    });
+    await test.step('Set pagination to 10 using top paginator', async () => {
+      await expect(page.locator('button#topPaginationWidgetId-top-toggle')).toHaveText(
+        '1 - 12 of 12',
+      );
+      await page.click('button#topPaginationWidgetId-top-toggle');
+      await page.getByRole('menuitem', { name: '10 per page' }).click();
+    });
+    await test.step('Assert there are only ten repos displayed', async () => {
+      await expect(page.locator('button#topPaginationWidgetId-top-toggle')).toHaveText(
+        '1 - 10 of 12',
```

Comment:
Could you here verify that there are only 10 repos on the screen? You can do that by doing some seletor and then `.toHaveCount(10)`

Reply:
Working on that

Reply:
Updated that section, now checks `.toHaveCount` in three places.